packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -620,24 +639,46 @@ export class RemotedFluidDataStoreContext extends FluidDataStoreContext {
                     throw new Error(`Invalid snapshot format version ${snapshotFormatVersion}`);
                 }
                 this.pkg = pkgFromSnapshot;
-                isRootStore = isRootDataStore;
             }
 
-            /**
-             * If there is no isRootDataStore in the attributes blob, set it to true. This will ensure that
-             * data stores in older documents are not garbage collected incorrectly. This may lead to additional
-             * roots in the document but they won't break.
-             */
             this.details = {
                 // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
                 pkg: this.pkg!,
-                isRootDataStore: isRootStore ?? true,
                 snapshot: tree ?? undefined,
+                // back-compat - 0.30. This has been moved to IGCDetails.
+                isRootDataStore: true,
             };
         }
 
         return this.details;
     }
+
+    protected async getInitialGCDetails(): Promise<IGCDetails> {
+        if (this.initialGcDetails === undefined) {
+            /**
+             * Default value of initial GC details in case the initial snapshot does not have GC details blob.
```

Comment:
I'm not sure this comment is correct (or reflects how I understand it works).
This info is used when data store is not realized, to keep system pay-per-play.
But if old file does not have that data, we can't just use empty gcNodes, we have to realize data store and generate that info, right?

Reply:
Yes, thank you, I missed that! :-).
I have updated it as follows:
- If GC blob in not available (for old data stores/channel contexts), initial GC nodes are set to undefined.
- Local data store returns initial GC nodes as undefined too.
- SummarizerNodeWithGC handles the above scenarios and summarizes even if nothing has changed to get GC nodes.